NullPointerException when analyzing source file

```
ERROR: Error during Sonar runner execution
org.sonar.runner.impl.RunnerException: Unable to execute Sonar
        at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:91)
        at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
        at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
        at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
        at org.sonar.runner.api.Runner.execute(Runner.java:100)
        at org.sonar.runner.Main.executeTask(Main.java:70)
        at org.sonar.runner.Main.execute(Main.java:59)
        at org.sonar.runner.Main.main(Main.java:53)
Caused by: org.sonar.squidbridge.api.AnalysisException: Unable to analyze file: D:\dev\bryce\src\Bryce\Plugins\RTGLibs\mrmgen\src\shared\Ijldefs.h
        at org.sonar.squidbridge.AstScanner.scanFiles(AstScanner.java:127)
        at org.sonar.plugins.cxx.squid.CxxSquidSensor.analyse(CxxSquidSensor.java:107)
        at org.sonar.batch.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:79)
        at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:70)
        at org.sonar.batch.phases.PhaseExecutor.execute(PhaseExecutor.java:119)
        at org.sonar.batch.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:194)
        at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:93)
        at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:78)
        at org.sonar.batch.scan.ProjectScanContainer.scan(ProjectScanContainer.java:233)
        at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:228)
        at org.sonar.batch.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:221)
        at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:93)
        at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:78)
        at org.sonar.batch.scan.ScanTask.scan(ScanTask.java:64)
        at org.sonar.batch.scan.ScanTask.execute(ScanTask.java:51)
        at org.sonar.batch.bootstrap.TaskContainer.doAfterStart(TaskContainer.java:125)
        at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:93)
        at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:78)
        at org.sonar.batch.bootstrap.BootstrapContainer.executeTask(BootstrapContainer.java:173)
        at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:95)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
        at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:48)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
        ... 9 more
Caused by: java.lang.NullPointerException
        at org.sonar.cxx.visitors.AbstractCxxPublicApiVisitor.visitEnumSpecifier(AbstractCxxPublicApiVisitor.java:414)
        at org.sonar.cxx.visitors.AbstractCxxPublicApiVisitor.visitNode(AbstractCxxPublicApiVisitor.java:149)
        at com.sonar.sslr.impl.ast.AstWalker.visitNode(AstWalker.java:114)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:85)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.visitChildren(AstWalker.java:99)
        at com.sonar.sslr.impl.ast.AstWalker.visit(AstWalker.java:87)
        at com.sonar.sslr.impl.ast.AstWalker.walkAndVisit(AstWalker.java:69)
        at org.sonar.squidbridge.AstScanner.scanFiles(AstScanner.java:106)
        ... 35 more
ERROR:
ERROR: Re-run SonarQube Runner using the -X switch to enable full debug logging.
```

Sonar claims my file is invalid, but the C++ compiler doesn't have any problems...

--- a/cxx-squid/src/main/java/org/sonar/cxx/visitors/AbstractCxxPublicApiVisitor.java
+++ b/cxx-squid/src/main/java/org/sonar/cxx/visitors/AbstractCxxPublicApiVisitor.java
@@ -84,7 +84,8 @@ public abstract class AbstractCxxPublicApiVisitor<GRAMMAR extends Grammar>
      */
     private static final boolean DUMP = false;
 
-    private static final String UNNAMED_CLASSIFIER_ID = "<unnamed>";
+    private static final String UNNAMED_CLASSIFIER_ID = "<unnamed class>";
+    private static final String UNNAMED_ENUM_ID       = "<unnamed enumeration>";
 
     public interface PublicApiHandler {
         void onPublicApi(AstNode node, String id, List<Token> comments);
@@ -412,17 +413,19 @@ public abstract class AbstractCxxPublicApiVisitor<GRAMMAR extends Grammar>
     }
 
     private void visitEnumSpecifier(AstNode enumSpecifierNode) {
+        AstNode enumIdNode = enumSpecifierNode.getFirstDescendant(
+                GenericTokenType.IDENTIFIER);
+
+        String enumId = (enumIdNode == null)?
+                UNNAMED_ENUM_ID : enumIdNode.getTokenValue();
+
         if (!isPublicApiMember(enumSpecifierNode)) {
-            logDebug(enumSpecifierNode.getFirstDescendant(
-                    GenericTokenType.IDENTIFIER).getTokenValue()
-                    + " not in public API");
+            logDebug(enumId + " not in public API");
             return;
         }
 
         visitPublicApi(
-                enumSpecifierNode,
-                enumSpecifierNode.getFirstDescendant(
-                        GenericTokenType.IDENTIFIER).getTokenValue(),
+                enumSpecifierNode, enumId,
                 getBlockDocumentation(enumSpecifierNode));
 
         // deal with enumeration values

--- a/cxx-squid/src/test/java/org/sonar/cxx/CxxPublicApiVisitorTest.java
+++ b/cxx-squid/src/test/java/org/sonar/cxx/CxxPublicApiVisitorTest.java
@@ -141,6 +141,11 @@ public class CxxPublicApiVisitorTest {
         testFile("src/test/resources/metrics/unnamed_class.h", 3, 1, false);
     }
 
+    @Test
+    public void unnamed_enum() {
+        testFile("src/test/resources/metrics/unnamed_enum.h", 1, 1, false);
+    }
+
     @SuppressWarnings("unchecked")
     @Test
     public void public_api() {
@@ -211,7 +216,7 @@ public class CxxPublicApiVisitorTest {
         expectedIdCommentMap.put("protectedClass", "protectedClass");
         expectedIdCommentMap.put("operator[]", "operator");
         expectedIdCommentMap.put("bitfield", "bitfield");
-        expectedIdCommentMap.put("<unnamed>", "<unnamed>");
+        expectedIdCommentMap.put("<unnamed class>", "<unnamed>");
         expectedIdCommentMap.put("testField2", "testField2");
         expectedIdCommentMap.put("testUnnamedStructVar", "testUnnamedStructVar");
 

--- a/cxx-squid/src/test/resources/metrics/unnamed_enum.h
+++ b/cxx-squid/src/test/resources/metrics/unnamed_enum.h
@@ -1,0 +1,3 @@
+enum /*anonymous enum*/ {
+};
+

